### ISSUE TITLE
Convergence difficulty in LCAO gamma only calculation when adding electrons in Silicon

### ISSUE THREAD

**Issue by hongriTianqi** [deepmodeling/abacus-develop#545]
## Describe the Bug
Convergence difficulty met in LCAO gamma only calculation when adding electrons in silicon single crystal calculation.

## To Reproduce
1. cat 311_NO_GO_elec_add/OUT.autotest/running_scf.log
2. cat 311_NO_GO_S2_elec_add/OUT.autotest/running_scf.log

[NO_GO_elec_add_err.tar.gz](https://github.com/deepmodeling/abacus-develop/files/7676365/NO_GO_elec_add_err.tar.gz)

**Comment by mohanchen** [comment]
We should have one person to test these cases. First of all, we should be clear about the formula to add or minus xx electrons. Then, we should have documents. Last, we should have a thorough test on both PW and LCAO cases.

**Comment by hongriTianqi** [comment]
This issue has not been addressed, so it should be kept open.
@Unboundwill @dyzheng @mohanchen

**Comment by wenfei-li** [comment]
I will take a look at this issue.

**Comment by wenfei-li** [comment]
It seems there is indeed some tricks for doing such kind of calculations. Sometimes it is hard for the SCF step to break the symmetry of up and down spin; in this case we can set an initial spin on one of the atoms.
As in this case, I was able to achieved SCF convergence with nspin = 2 and spin 1.0 placed on the first Si atom. However, the convergence is indeed hard to obtain as it took around 100 SCF steps.
I have also tested the case using Quantum Espresso pwscf, and it also took some time for SCF to converge. Difficulty in convergence seems to be a common issue for spin polarized calculations and not particular to ABACUS.
There does not seem to be a bug and I don't think we have plan as of now to address the issue. As a result, I'll close it.
